latch/registry/table.py

```diff
@@ -0,0 +1,290 @@
+import asyncio
+import json
+from abc import abstractmethod
+from dataclasses import dataclass, field
+from typing import Any, Dict, Iterator, List, Optional, Type, TypedDict
+
+import gql
+import graphql.language as l
+import graphql.type as t
+import graphql.utilities as u
+
+from latch.gql.execute import execute, get_transport
+from latch.registry.record import Record
+from latch.registry.types import (
+    RegistryDBValue,
+    RegistryInvalidValue,
+    registry_empty_cell,
+)
+from latch.registry.utils import (
+    RegistryTransformerException,
+    to_python_literal,
+    to_registry_literal,
+)
+
+
+@dataclass(frozen=True)
+class ListRecordsOutput:
+    records: List[Record]
+    errors: List[Exception]
+
+
+@dataclass
+class Table:
+    id: str
+
+    def __post_init__(self):
+        self._display_name: Optional[str] = None
+        self._columns: Optional[List[Dict[str, Dict]]] = None
+        self._data: Optional[Dict[str, Dict]] = None
+
+    def load(self):
+        self._load()
+
+    def _load(self):
+        if self._data is not None:
+            return self._data
+
+        # todo(ayush): paginate column defs too?
+        self._data = execute(
+            """
+            query tableQuery ($argTableId: BigInt!) {
+                catalogExperiment(id: $argTableId) {
+                    id
+                    displayName
+                    catalogExperimentColumnDefinitionsByExperimentId {
+                        nodes {
+                            key
+                            type
+                        }
+                    }
+                }
+            }
+            """,
+            variables={"argTableId": self.id},
+        )
+        self._display_name = self._data["catalogExperiment"]["displayName"]
+        self._columns = self._data["catalogExperiment"][
+            "catalogExperimentColumnDefinitionsByExperimentId"
+        ]["nodes"]
+
+        return self._data
```

Comment:
I don't think the return value is used anywhere
if it isn't, remove it from here and rename the function to `def load(self)` without an `_` since that's the only difference right now